float64 p_in
float64 v_in
float64 kp_in
float64 kd_in
float64 t_in